lib/badge-data.js

```diff
@@ -39,8 +52,12 @@ function isHexColor (s = ''){
 function makeColor(color) {
   if (isHexColor(color)) {
     return '#' + color;
-  } else {
+  } else if (colorschemes[color] !== undefined){
+    return colorschemes[color].colorB;
+  } else if (isCSSColor(color)){
     return color;
+  } else {
+    return undefined;
```

Comment:
Is this change related to simpleicons?

Reply:
Kind of, I wasn't sure whether to make a new function for the color or just modify this function as it seemed to still fit quite well, I think it makes more sense to only return a valid color rather than just assume anything that's not hex is a valid color/colorscheme, and the current test were expecting any non valid color to return undefined.

Also there wasn't any test for this function so i've added these too, which are very similar to `setBadgeColor()`.

The main differences between `setBadgeColor()` and `makeColor()` are:
### `setBadgeColor()`:
will set the either `badgeData.colorB` or `badgeData.colorscheme`_(which i think may be redundant now, but will look at that in another PR)_ to the desired color.
### `makeColor()`:
returns a valid color.